### PR TITLE
[front] featue: add tables list endpoint in dataSourceView

### PR DESCRIPTION
**Pull request by tdraier**
## Description

replace usage of useDataSourceTables with useDataSourceViewTables in dust apps, add corresponding endpoint

## Risk

none

## Deploy Plan

deploy `front`

### CODE REVIEW

File: front/pages/api/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/tables/index.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+const DEFAULT_LIMIT = 100;
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<ListTablesResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  const dataSourceView = await DataSourceViewResource.fetchById(
+    auth,
+    req.query.dsvId as string
```
Comment (flvndvd) [inline front/pages/api/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/tables/index.ts:27]: No cast, let's do:
```
const {dsvId} = req.query;

if (typeof dsvId !== "string") {
  // TODO: Handle error.
} 
```

File: front/pages/api/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/tables/index.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      }
+
+      const pagination = paginationRes.value;
+
+      const contentNodes = await getContentNodesForDataSourceView(
+        dataSourceView,
+        {
+          includeChildren: false,
+          viewType: "tables",
+          onlyCoreAPI: true,
```
Comment (flvndvd) [inline front/pages/api/w/[wId]/vaults/[vId]/data_source_views/[dsvId]/tables/index.ts:67]: Let's add a small comment around this line 🙏

**Review by flvndvd — APPROVED** [review]

File: front/lib/swr/data_source_views.ts [hunk 3]
```diff
@@ -211,3 +212,28 @@ export function useDataSourceViewConnectorConfiguration({
...
+}: {
+  dataSourceView: DataSourceViewType | null;
+  workspaceId: string;
+}) {
+  const tablesFetcher: Fetcher<ListTablesResponseBody> = fetcher;
+  const disabled = !dataSourceView;
+
+  const { data, error, mutate } = useSWRWithDefaults(
+    disabled
+      ? null
```
Comment (flvndvd) [inline front/lib/swr/data_source_views.ts:228]: @Fraggle does this follow the new approach?

Comment (Fraggle) [reply front/lib/swr/data_source_views.ts:228]: If `dataSourceView` has to be nullable, yes. Passing `{disabled}` as an option wouldn't change anything (mutate would not work etc..)

File: front/lib/swr/data_source_views.ts [hunk 4]
```diff
@@ -211,3 +212,28 @@ export function useDataSourceViewConnectorConfiguration({
...
+      : `/api/w/${workspaceId}/vaults/${dataSourceView.vaultId}/data_source_views/${dataSourceView.sId}/tables`,
+    tablesFetcher
+  );
+
+  return {
+    tables: useMemo(() => (data ? data.tables : []), [data]),
+    isTablesLoading: !error && !data,
```
Comment (Fraggle) [inline front/lib/swr/data_source_views.ts:235]: isTablesLoading is going to return `true` when disabled, i suggest `!disabled && !error && !data`

**Pull request merged**